eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -526,20 +549,91 @@ public void testDifferentCacheFilesPerKey() throws IOException {
         .write(Configuration.builder(jsonBytes, true).build().serializeFlagConfigToBytes());
 
     // Initialize with offline mode to prevent instance2 from pulling config via fetch.
-    initClient(TEST_HOST, true, false, false, true, null, null, DUMMY_OTHER_API_KEY, true, null);
+    initClient(
+        TEST_HOST, true, false, false, true, null, null, DUMMY_OTHER_API_KEY, true, null, false);
 
     // Ensure API key 2 uses its cache
     double apiKey2Assignment =
         EppoClient.getInstance().getDoubleAssignment("numeric_flag", "alice", 0.0);
     assertEquals(1.2345, apiKey2Assignment, 0.0000001);
 
     // Reinitialize API key 1 to be sure it used its cache
-    initClient(TEST_HOST, true, false, false, true, null, null, DUMMY_API_KEY, false, null);
+    initClient(TEST_HOST, true, false, false, true, null, null, DUMMY_API_KEY, false, null, false);
     // API Key 1 will fetch and then populate its cache with the usual test data
     apiKey1Assignment = EppoClient.getInstance().getDoubleAssignment("numeric_flag", "alice", 0.0);
     assertEquals(3.1415926, apiKey1Assignment, 0.0000001);
   }
 
+  @Test
+  public void testForceIgnoreCache() throws IOException, ExecutionException, InterruptedException {
+    cacheUselessConfig();
+    // Initialize with "useless" cache available.
+    new EppoClient.Builder(DUMMY_API_KEY, ApplicationProvider.getApplicationContext())
+        .host(TEST_HOST)
+        .assignmentLogger(mockAssignmentLogger)
+        .obfuscateConfig(true)
+        .forceReinitialize(true)
+        .offlineMode(false)
+        .buildAndInitAsync()
+        .get();
+    double assignment = EppoClient.getInstance().getDoubleAssignment("numeric_flag", "alice", 0.0);
+    assertEquals(0.0, assignment, 0.0000001);
+
+    // Initialize again with "useless" cache available but ignoreCache = true
+    cacheUselessConfig();
+    new EppoClient.Builder(DUMMY_API_KEY, ApplicationProvider.getApplicationContext())
+        .host(TEST_HOST)
+        .assignmentLogger(mockAssignmentLogger)
+        .obfuscateConfig(true)
+        .forceReinitialize(true)
+        .offlineMode(false)
+        .ignoreCache(true)
+        .buildAndInitAsync()
+        .get();
+
+    double properAssignment =
+        EppoClient.getInstance().getDoubleAssignment("numeric_flag", "alice", 0.0);
+    assertEquals(3.1415926, properAssignment, 0.0000001);
+  }
+
+  private void cacheUselessConfig() throws IOException {
```

Comment:
minor, but I'd prefer to avoid `throws` signature and just try catch within here and rethrow as a runtime exception

Reply:
✔️